README.md

```diff
@@ -18,4 +18,129 @@ Also you have to configure the `github` maven repository. You can either include
     <id>github</id>
     <url>https://andrsuh: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@maven.pkg.github.com/andrsuh/tiny-event-sourcing</url>
 </repository>
-```
\ No newline at end of file
+```
+# Example of how to use library
+## Theory
+https://www.eventstore.com/event-sourcing - a good source of information about event sourcing and CQRS.
+## Example
+First, when we implement events sourcing pattern we have to define aggregates. In this example we will be having user aggregate.
+```kotlin
+@AggregateType(aggregateEventsTableName = "aggregate-user")
+data class UserAggregate(
+    override val aggregateId: String
+) : Aggregate {
+    override var createdAt: Long = System.currentTimeMillis()
+    override var updatedAt: Long = System.currentTimeMillis()
+
+    var userName: String = ""
+    var userLogin: String = ""
+    var userPassword: String = ""
+    lateinit var defaultPaymentId: UUID
+    lateinit var defaultAddressId: UUID
+    var paymentMethods = mutableMapOf<UUID, PaymentMethod>()
+    var deliveryAddresses = mutableMapOf<UUID, DeliveryAddress>()
+}
+```
+@AggregateType annotations makes library understand that this is the aggregate recognisable by library.
+Each aggregate has its own domain events. They're written in UserAggregateDomainEvents.kt.
+This is the example of one of the events:
+```kotlin
+@DomainEvent(name = USER_CREATED_EVENT)
+class UserCreatedEvent(
+    val userLogin: String,
+    val userPassword: String,
+    val userName: String,
+    val userId: String,
+    createdAt: Long = System.currentTimeMillis(),
+) : Event<UserAggregate>(
+    name = USER_CREATED_EVENT,
+    createdAt = createdAt,
+    aggregateId = userId,
+) {
+    override fun applyTo(aggregate: UserAggregate) {
+        aggregate.userLogin = userLogin
+        aggregate.userPassword = userPassword
+        aggregate.userName = userName
+    }
+}
+```
+@DomainEvent annotation is also from the library. It accepts name of the event.
+Each event accepts some parameters from the constructor, returns event that belongs to the aggregate:
+```kotlin
+Event<UserAggregate>
+```
+and has override method applyTo that does manipulations with aggregate state.
+This is how this event looks in database:
+![](Example1.png)
+Second important concept of this example are subscribers. We can do subscriptions 2 ways.
+First one is this:
+```kotlin
+@Service
+@AggregateSubscriber(aggregateClass = UserAggregate::class, subscriberName = "demo-user-stream")
+class AnnotationBasedUserEventsSubscriber {
+    val logger: Logger = LoggerFactory.getLogger(AnnotationBasedUserEventsSubscriber::class.java)
+
+    @SubscribeEvent
+    fun userCreatedSubscriber(event: UserCreatedEvent) {
+        logger.info("User created {}", event.userName)
+    }
+}
+```
+As can be seen from the name of the class, it's annotation based. 
```

Comment:
mettion that we can lister to all the aggregate events but also can only take and handle some of them by defining typed handlers and apply logic based on event type

Reply:
take from JavaDoc